packages/xo-server-sdn-controller/src/index.js

```diff
@@ -406,124 +409,165 @@ class SDNController extends EventEmitter {
       },
     })
 
-    // FIXME: we should monitor when xapis are added/removed
-    this._xapis = this._xo.getAllXapis()
-    await Promise.all(
-      map(this._xapis, async xapi => {
-        await xapi.objectsFetched
+    forOwn(this._xo.getAllXapis(), xapi => {
+      if (xapi.status === 'connected') {
+        this._xapiConnected(xapi)
```

Comment:
A method must be a verb because it do an action. I suggest `handleConnectedXapi`.